### PR TITLE
Setup heroku pipeline  integration

### PR DESCRIPTION
**Pull request by kabaros**
Migrates from using postgres on AWS to using the Heroku add-on.

- adds config to `app.json` to provision the add-on in review apps, 
- adds a release script to setup the DB before each release
- updates knex config to use heroku postgres `DATABASE_URL`
- cleans up environment variables and CircleCI config
- adds linting script to the project. Run `npm run lint` before committing any code, and fix any errors.

### CODE REVIEW

File: server/knexfile.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   development: {
-    client: "postgresql",
+    client: "pg",
```
Comment (djgrant) [inline server/knexfile.js]: Not sure if I did this my PR too but there's both `pg` and `postgresql`

File: .circleci/config.yml [hunk 2]
```diff
@@ -11,6 +11,7 @@ jobs:
     steps:
       - checkout
       - run: npm --prefix client install
+      - run: npm --prefix client run lint
```
Comment (djgrant) [inline .circleci/config.yml]: would this be better in a precommit hook using something like https://www.npmjs.com/package/pre-commit?

Comment (djgrant) [reply .circleci/config.yml]: Actually, this is fine. I was thinking this was running lint with auto save on

Comment (kabaros) [reply .circleci/config.yml]: I will try those .. I remember having lots of issues with `pre-commit` and `husky` on windows though

**Pull request merged**